src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CurrentPeriodServiceImpl.java

```diff
@@ -28,6 +36,24 @@ public void addKind(CurrentPeriod rest) {
 
     @Override
     public String getResourceName() {
-        return "current-period";
+        return ResourceName.CURRENT_PERIOD.getName();
+    }
+
+    @Override
+    public String createSelfLink(Transaction transaction, String companyAccountId) {
+        return transaction.getLinks().get(LinkType.SELF.getLink()) + "/company-account/"
```

Comment:
we should be using `ResourceName.COMPANY_ACCOUNT.getName()` and `ResourceName.SMALL_FULL.getName()` here rather than hardcoding these strings